includes/wikia/parser/templatetypes/TemplateTypesParser.class.php

```diff
@@ -82,13 +82,15 @@ public static function onGetTemplateDom( $title, $args, $frame, &$outputText ) {
 	public static function onEndBraceSubstitution( $templateTitle, &$templateWikitext ) {
 		wfProfileIn( __METHOD__ );
 
-		if ( ContextLinkTemplate::shouldTemplateBeProcessed( $templateWikitext ) ) {
+		if ( self::shouldTemplateBeParsed() && !empty( $templateWikitext ) && !TemplateArgsHelper::containsUnexpandedArgs( $templateWikitext ) ) {
```

Comment:
I thing that you may want to leave the existing structure in place, rather extracting methods to simplify the flow of this function, i.e. 
1. Determine whether the template should be parsed
2. If yes, fetch template type
3. Switch() over the type, checking type-specific processing conditions and invoking the right handle function